vowpalwabbit/reductions/epsilon_decay.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#pragma once
+
+#include "array_parameters.h"
+#include "distributionally_robust.h"
+#include "reductions_fwd.h"
+#include "scored_config.h"
+
+namespace VW
+{
+namespace epsilon_decay
+{
+VW::LEARNER::base_learner* epsilon_decay_setup(VW::setup_base_i&);
+
+struct epsilon_decay_score : scored_config
+{
+  epsilon_decay_score() = default;
+  epsilon_decay_score(double alpha, double tau, uint64_t model_idx)
+      : VW::scored_config(alpha, tau), _model_idx(model_idx)
+  {
+  }
+  float get_upper_bound() const { return this->current_ips(); }
+  float get_lower_bound() const { return _lower_bound; }
+  uint64_t get_model_idx() const { return _model_idx; }
+  void update_bounds(float w, float r);
+
+  float _lower_bound = 0.f;
```

Comment:
private variables?

Reply:
these are accessed directly so cannot be private